### PR TITLE
Add C++ stacktrace commands for Linux to dev-docs

### PR DESCRIPTION
**Pull request by DanielMurphy22**
**Description of work.**

Added 2 new tools to UserSupport troubleshooting advice on dev-docs

**To test:**

Build dev-docs-html and Code review 

*There is no associated issue.*

*This does not require release notes* because **dev docs update**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: dev-docs/source/UserSupport.rst [hunk 1]
```diff
@@ -205,8 +205,17 @@ For a full release, ``/opt/Mantid/`` is likely the correct install path. Take ca
 
 11. Further diagnosis for process monitoring: `strace <https://strace.io/>`_.
 
+12. To get a stacktrace from C++ follow these steps:
```
Comment (jhaigh0) [inline dev-docs/source/UserSupport.rst:208]: The bullet points will format more nicely if you bring them forward two spaces, add a new line, and keep each one on one line. Like this:
```
12. To get a stacktrace from C++ follow these steps:

    - Before opening MantidWorkbench, open a terminal and run ``ulimit -c unlimited``
    - Then in the same terminal, open MantidWorkbench with ``/opt/mantidworkbench/bin/mantidworkbench`` and use as normal.
    - If there is a crash, then you will find a ``core.1234`` (some number) file in the current working directory of the terminal, which the user can send to us!
    - Run ``gdb python3 core.1234`` to get the stacktrace from this coredump file.
```

This is alwasy an anoying thing with rst.

**Review by jhaigh0 — CHANGES_REQUESTED** [review]

File: dev-docs/source/UserSupport.rst (hunk 1 above)
Comment (jhaigh0) [reply dev-docs/source/UserSupport.rst:208]: Ah okay I think you want to move them forward four spaces not, so the hyphen is inline with the the T of 'To get a stacktrace...'. Then it'll indent them without the weird formatting.

File: dev-docs/source/UserSupport.rst [hunk 2]
```diff
@@ -205,8 +205,18 @@ For a full release, ``/opt/Mantid/`` is likely the correct install path. Take ca
...
+
+- Before opening MantidWorkbench, open a terminal and run ``ulimit -c unlimited``
+- Then in the same terminal, open MantidWorkbench with ``/opt/mantidworkbench/bin/mantidworkbench`` and use as normal.
+- If there is a crash, then you will find a ``core.1234`` (some number) file
+  in the current working directory of the terminal, which the user can send to us!
+- Run ``gdb python3 core.1234`` to get the stacktrace from this coredump file.
```
Comment (jhaigh0) [inline dev-docs/source/UserSupport.rst:214]:
```suggestion
    - Before opening MantidWorkbench, open a terminal and run ``ulimit -c unlimited``
    - Then in the same terminal, open MantidWorkbench with ``/opt/mantidworkbench/bin/mantidworkbench`` and use as normal.
    - If there is a crash, then you will find a ``core.1234`` (some number) file
      in the current working directory of the terminal, which the user can send to us!
    - Run ``gdb python3 core.1234`` to get the stacktrace from this coredump file.
```

Comment (DanielMurphy22) [reply dev-docs/source/UserSupport.rst]: Thanks, this is exactly what I wanted! the next step 13 was a little close to the end of the list so I've added some pipe characters to help with spacing!

Comment (jhaigh0) [reply dev-docs/source/UserSupport.rst]: The pipe characters seem to un-indent bullet point 13 but I do think it looks better regardless with that extra space. I've had a go at finding a way to indent properly with the same spacing but have had no luck.

Comment (DanielMurphy22) [reply dev-docs/source/UserSupport.rst]: I've also played around with all the options and I've now just swapped the order of 12 and 13 so it looks right! I have also added pipe characters before each OS section so tell me if you think that renders well or not!

**Review by jhaigh0 — APPROVED** [review]
All the formatting works as intended now 👍

**Pull request merged**